gr-blocks/python/blocks/qa_message_strobe.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2021 Free Software Foundation, Inc.
+#
+# This is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3, or (at your option)
+# any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software; see the file COPYING.  If not, write to
+# the Free Software Foundation, Inc., 51 Franklin Street,
+# Boston, MA 02110-1301, USA.
+#
+
+from gnuradio import gr, gr_unittest
+from gnuradio import blocks
+import pmt
+import time
+
+# this test tests message strobe and message debug blocks against each other
+class qa_message_strobe(gr_unittest.TestCase):
+
+    def setUp(self):
+        self.tb = gr.top_block()
+
+    def tearDown(self):
+        self.tb = None
+
+    def test_001_t(self):
+        test_str = "test_msg"
+        new_msg = "new_msg"
+        period_ms = 7
+        msg_strobe = blocks.message_strobe(pmt.intern(test_str), period_ms)
+        msg_debug = blocks.message_debug()
+
+        self.tb.msg_connect(msg_strobe, "strobe", msg_debug, "store")
+
+        self.tb.start()
+
+        self.assertTrue(msg_debug.num_messages() == 0) # 1st call, expect 0
+        time.sleep(0.01) # floor(10/7) = 1
+        self.assertTrue(msg_debug.num_messages() == 1) # 2nd call == 1
+        time.sleep(0.01) # floor(20/7) = 2
+        self.assertTrue(msg_debug.num_messages() == 2) # 3rd call == 2
+        time.sleep(0.01) # floor(30/7) = 4 messages
```

Comment:
Hi, yes I agree, those numbers are too tight. I experienced it yesterday evening once when my system was under heavy load but did not manage to update it quickly then. 
I am a bit hestitent to increase the sleep time to over a second, but I increased it by a factor >10 and added a margin of one message to the test.
Do you think that suffices?
(will push it in a second)
Also changed the assertTrue statements to assertEqual/assertAlmostEqual as they provide more information on failure.

Reply:
For the QA, it is critical that we don't introduce tests that fail intermittently.  I like these tests, but agree that the timing needs to be loosened up so there is no chance of failure.  For these tests, we just want to make sure the message passing mechanisms functionally work, and if this takes a couple of seconds, it is ok.

Reply:
Hi @mormj  , thank you as well for the feedback! 
Ok, one more order of magnitude :D

how about following test procedure:
- start
- test if messages in block is equal 1, with 2 messages tolerance (in case it was really fast to send the first msgs and we were blocked)
- wait for 1 sec
- test number messages equal 10 with 3 messages tolerance
- sleep 1 more second
- test equal 20 messages, again 3 msgs tolerance
- update message string
- wait 1 second
- check that first message received is equal to initial text string
- check that last message received is equal to updated string

what do you think of the margins now? (and if the test is meaningful enough)

Cheers!

Reply:
i think this is meaningful enough to show the functionality of the message strobe and perform validation of the msgport connections - the margins seem sufficient to where if it breaks, it should fail this test.

Reply:
hey, I modified the test procedure as written above. (tests still pass locally for me as expected).
hope this will do :-)